cmd/plume/prune.go

```diff
@@ -204,11 +256,45 @@ func pruneAWS(ctx context.Context, spec *channelSpec) {
 					}
 					board := fmt.Sprintf("%s-usr", arch)
 					var version string
+					var softDeleteDate string
 					for _, t := range image.Tags {
 						if *t.Key == "Version" {
 							version = *t.Value
 						}
+						if *t.Key == "SoftDeleteDate" {
+							softDeleteDate = *t.Value
+						}
 					}
+					if softDeleteDate == "" && daysSoftDeleted != 0 {
+						softDeleteDate = now.Format(time.RFC3339)
+						// remove LaunchPermission
+						_, err = api.RemoveLaunchPermission(*image.ImageId)
+						if err != nil {
+							plog.Fatalf("Error removing launch permission from %v: %v", *image.Name, err)
+						}
+						// add tag
+						err = api.CreateTags([]string{*image.ImageId}, map[string]string{"SoftDeleteDate": softDeleteDate})
+						if err != nil {
+							plog.Fatalf("Error adding tag to %v: %v", *image.Name, err)
+						}
+						plog.Infof("Image %v has been soft-deleted", *image.Name)
+						stats.SoftDeleted += 1
+						continue
+					} else if daysSoftDeleted != 0 {
+						// check if the image is still soft-deleted
+						softdeletedate, err := time.Parse(time.RFC3339, softDeleteDate)
```

Comment:
camelCase, please. Likely need a different name, as you have `softdeletedate` and `softDeleteDate`.

Reply:
Camelcased and renamed to `softDeleteDateTs`